controllers/managedfusionoffering_controller.go

```diff
@@ -0,0 +1,89 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	v1alpha1 "github.com/red-hat-storage/managed-fusion-agent/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	controller "sigs.k8s.io/controller-runtime/pkg/controller"
+)
+
+// ManagedFusionOfferingReconciler reconciles a ManagedFusionOffering object
+type ManagedFusionOfferingReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+
+	ctx                   context.Context
+	managedFusionOffering *v1alpha1.ManagedFusionOffering
+}
+
+//+kubebuilder:rbac:groups=misf.ibm.com,resources=managedfusionofferings,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=misf.ibm.com,resources=managedfusionofferings/status,verbs=get;update;patch
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *ManagedFusionOfferingReconciler) SetupWithManager(mgr ctrl.Manager, ctrlOptions *controller.Options) error {
+	managedFusionOfferingController := ctrl.NewControllerManagedBy(mgr).
```

Comment:
This is not a controller, it is a controller builder. The controller is returned only after you call the complete method.
Also, no need to mention managedFusion as this is building a single controller under the context of this `SetupWithManager` call. 

. 
```suggestion
	controllerBuilder  := ctrl.NewControllerManagedBy(mgr).
```